tests/unit/lms/services/canvas_api_test.py

```diff
@@ -198,6 +198,74 @@ def old_oauth2_token(self, db_session, pyramid_request):
         return old_oauth2_token
 
 
+class TestAuthenticatedUsersSections:
+    """Unit tests for CanvasAPIClient.authenticated_users_sections()."""
+
+    @pytest.mark.usefixtures("access_token")
+    def test_it(
+        self,
+        ai_getter,
+        canvas_api_client,
+        CanvasAPIHelper,
+        canvas_api_helper,
+        send_with_refresh_and_retry,
+        CanvasAuthenticatedUsersSectionsResponseSchema,
+        pyramid_request,
+    ):
+        sections = canvas_api_client.authenticated_users_sections("test_course_id")
+
+        # It initializes canvas_api_helper.
+        CanvasAPIHelper.assert_called_once_with(
+            pyramid_request.lti_user.oauth_consumer_key,
+            ai_getter,
+            pyramid_request.route_url,
+        )
+
+        # It gets the prepared request.
+        canvas_api_helper.authenticated_users_sections_request.assert_called_once_with(
+            "test_access_token", "test_course_id"
+        )
+
+        # It sends the prepared request.
+        send_with_refresh_and_retry.assert_called_once_with(
+            canvas_api_helper.authenticated_users_sections_request.return_value,
+            CanvasAuthenticatedUsersSectionsResponseSchema,
+            "test_refresh_token",
+        )
+
+        # It returns the sections.
+        assert sections == send_with_refresh_and_retry.return_value
```

Comment:
Great big integration test here because the method under test is an integration method that does nothing other than integrate a few other things together. It'd be nicer if the code design was such that an integration method wasn't needed, or at least was much smaller, then this integration test might not be needed or could be much simpler. Again, though, resisting the urge to redesign at this point